File: java/amp-bus/src/main/java/amp/bus/ITransportProvider.java
package amp.bus;


import cmf.bus.Envelope;
import cmf.bus.IDisposable;
import cmf.bus.IRegistration;


public interface ITransportProvider extends IDisposable {

    void onEnvelopeReceived(IEnvelopeReceivedCallback callback);

    void register(IRegistration registration) throws Exception;

    void send(Envelope envelope) throws Exception;

    void unregister(IRegistration registration) throws Exception;
}

File: dotnet/source/amp.bus/ITransportProvider.cs
using System;
using cmf.bus;


namespace amp.bus
{
    public interface ITransportProvider : IDisposable
    {

        //TODO: Why does ITransportProvider define an OnEnvelopeReceived and IEnvelopeReceiver does not?
        event Action<IEnvelopeDispatcher> OnEnvelopeReceived;


        void Send(Envelope env);

        void Register(IRegistration registration);

        void Unregister(IRegistration registration);
    }
}
